app/helpers.rb

```diff
@@ -31,4 +31,17 @@ def js_to_rb_date(date)
     Date.parse(date)
   end
 
+  def decline_requests(dates, space)
+    requests = Request.all(space_id: space.id, status: 1)  
+    date_conflicts = []
+    requests.each do | request |
+      requested_dates = get_date_range(request.start_date, request.end_date)
+      dates.each { |date| date_conflicts << date if requested_dates.include?(date) }
```

Comment:
repeats code in `date_match` method, so maybe extract the few `date_conflicts` lines into its own method and keep it dry?